### ISSUE TITLE
Stream specifier ':v' in filtergraph ... matches no streams.

### ISSUE THREAD

**Issue by alvaromlg** [moviemasher/moviemasher.rb#9]
Hello,

First of all, your app is awesome!

When I first launch the app I manage to transcode a video and trim it, but then if I keep playing with the trim/duration values I end getting this error and I am unable to use the transcoder again.

I cannot even transcode a former mash I successfully transcoded.

This is the error:

> Stream specifier ':v' in filtergraph description color=color=0x000000:duration=68.7:size=512x288:rate=30[layer0];[0:v]trim=duration=68.7,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0 matches no streams.

This is the ffmpeg pipeline with the error:
```
ffmpeg -y  -i /tmp/moviemasher/download/f2cc4616384daf5ebf4f4c960fbe842ebeffbb6673179590b4d5ae48df460b5d/downloaded-intermediate.wav -i /tmp/moviemasher/download/f2cc4616384daf5ebf4f4c960fbe842ebeffbb6673179590b4d5ae48df460b5d/downloaded.mp4 -filter_complex "color=color=0x000000:duration=68.7:size=512x288:rate=30[layer0];[0:v]trim=duration=68.7,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0" -af \[0:a\]atrim\=start\=0.0:duration\=68.7,asetpts\=expr\=PTS-STARTPTS -async 1 -t 68.7 -b:a 224k -r:a 44100 -c:a aac -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\=New\ Untitled\ Mash -pass 2 -passlogfile /tmp/moviemasher/render/0b669e98-30b7-422b-9eed-9307e9605845/pass-754c2298-fff1-4b8a-be89-39065e9a3395 /tmp/moviemasher/render/0b669e98-30b7-422b-9eed-9307e9605845/0ca2692b-d8c4-4292-80d3-47e4151e59bc.mp4
W, [2016-07-13T10:35:10.841954 #1]  WARN -- : two pass encoding failed, retrying in one
D, [2016-07-13T10:35:10.842017 #1] DEBUG -- : ffmpeg -y  -i /tmp/moviemasher/download/f2cc4616384daf5ebf4f4c960fbe842ebeffbb6673179590b4d5ae48df460b5d/downloaded-intermediate.wav -i /tmp/moviemasher/download/f2cc4616384daf5ebf4f4c960fbe842ebeffbb6673179590b4d5ae48df460b5d/downloaded.mp4 -filter_complex "color=color=0x000000:duration=68.7:size=512x288:rate=30[layer0];[0:v]trim=duration=68.7,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0" -af \[0:a\]atrim\=start\=0.0:duration\=68.7,asetpts\=expr\=PTS-STARTPTS -async 1 -t 68.7 -b:a 224k -r:a 44100 -c:a aac -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\=New\ Untitled\ Mash /tmp/moviemasher/render/0b669e98-30b7-422b-9eed-9307e9605845/0ca2692b-d8c4-4292-80d3-47e4151e59bc.mp4
D, [2016-07-13T10:35:10.906095 #1] DEBUG -- : ffmpeg version 3.0.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.9.2 (Debian 4.9.2-10)
  configuration: --enable-frei0r --enable-gpl --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libxvid --enable-postproc --enable-pthreads --enable-version3 --enable-zlib --extra-cflags=-I/usr/local/include/openjpeg
  libavutil      55. 17.103 / 55. 17.103
  libavcodec     57. 24.102 / 57. 24.102
  libavformat    57. 25.100 / 57. 25.100
  libavdevice    57.  0.101 / 57.  0.101
  libavfilter     6. 31.100 /  6. 31.100
  libswscale      4.  0.100 /  4.  0.100
  libswresample   2.  0.101 /  2.  0.101
  libpostproc    54.  0.100 / 54.  0.100
Guessed Channel Layout for  Input Stream #0.0 : stereo
Input #0, wav, from '/tmp/moviemasher/download/f2cc4616384daf5ebf4f4c960fbe842ebeffbb6673179590b4d5ae48df460b5d/downloaded-intermediate.wav':
  Metadata:
    encoder         : Lavf57.25.100
  Duration: 00:01:08.80, bitrate: 1411 kb/s
    Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 44100 Hz, 2 channels, s16, 1411 kb/s
Input #1, mov,mp4,m4a,3gp,3g2,mj2, from '/tmp/moviemasher/download/f2cc4616384daf5ebf4f4c960fbe842ebeffbb6673179590b4d5ae48df460b5d/downloaded.mp4':
  Metadata:
    major_brand     : mp42
    minor_version   : 0
    compatible_brands: isommp42
    creation_time   : 2015-04-21 06:31:12
  Duration: 00:01:08.80, start: 0.000000, bitrate: 2930 kb/s
    Stream #1:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 2735 kb/s, 25 fps, 25 tbr, 25 tbn, 50 tbc (default)
    Metadata:
      creation_time   : 1970-01-01 00:00:00
      handler_name    : VideoHandler
    Stream #1:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 191 kb/s (default)
    Metadata:
      creation_time   : 2015-04-21 06:31:13
      handler_name    : IsoMedia File Produced by Google, 5-11-2011
Stream specifier ':v' in filtergraph description color=color=0x000000:duration=68.7:size=512x288:rate=30[layer0];[0:v]trim=duration=68.7,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0 matches no streams.
```

I notice the input 0, which is the downloaded-intermediate.wav file I upload (in this case is a mp4) has no video, only sound.

When I do a ffprobe I get:
```
ffprobe downloaded-intermediate.wav
ffprobe version 3.0.2 Copyright (c) 2007-2016 the FFmpeg developers
  built with gcc 4.9.2 (Debian 4.9.2-10)
  configuration: --enable-frei0r --enable-gpl --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libxvid --enable-postproc --enable-pthreads --enable-version3 --enable-zlib --extra-cflags=-I/usr/local/include/openjpeg
  libavutil      55. 17.103 / 55. 17.103
  libavcodec     57. 24.102 / 57. 24.102
  libavformat    57. 25.100 / 57. 25.100
  libavdevice    57.  0.101 / 57.  0.101
  libavfilter     6. 31.100 /  6. 31.100
  libswscale      4.  0.100 /  4.  0.100
  libswresample   2.  0.101 /  2.  0.101
  libpostproc    54.  0.100 / 54.  0.100
Input #0, wav, from 'downloaded-intermediate.wav':
  Metadata:
    encoder         : Lavf57.25.100
  Duration: 00:01:08.80, bitrate: 1411 kb/s
    Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 44100 Hz, 2 channels, s16, 1411 kb/s
```

Is tihs intendeed? the filtergraph is trying to grab 0:v which is video and this file has only audio, although with 0:a it doesnt work neither.

**Comment by syntropo** [comment]
That is strange behavior... The [0:v] refers to the first video channel from the inputs, not simply the first input. So the command looks okay to me and should be assigning the mp4 file. This file seems to be half a decade old though - does the mash work with a more modern file encoding? Does the mash work if you add another visual layer to it? This would effectively reencode the video portion in order to merge it.

**Comment by alvaromlg** [comment]
Hello, thanks for your response.

I am not very good in ffmpeg so thanks for carlifying that for me. mm what do you mean with a more modern file encoding? Where does it appear my file has an old encoding? I cannot see it in the ffprobe

**Comment by alvaromlg** [comment]
Ok I re encoded my .mp4 with this ffmpeg and this pipeline:

```
ffmpeg version 2.8.6 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 5.1.1 (GCC) 20150618 (Red Hat 5.1.1-4)
  configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic' --enable-bzlib --disable-crystalhd --enable-frei0r --enable-gnutls --enable-ladspa --enable-libass --enable-libcdio --enable-libdc1394 --disable-indev=jack --enable-libfreetype --enable-libgsm --enable-libmp3lame --enable-openal --enable-libopencv --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libv4l2 --enable-libvpx --enable-libx264 --enable-libx265 --enable-libxvid --enable-x11grab --enable-avfilter --enable-avresample --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-runtime-cpudetect
  libavutil      54. 31.100 / 54. 31.100
  libavcodec     56. 60.100 / 56. 60.100
  libavformat    56. 40.101 / 56. 40.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 40.101 /  5. 40.101
  libavresample   2.  1.  0 /  2.  1.  0
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  2.101 /  1.  2.101
  libpostproc    53.  3.100 / 53.  3.100
Hyper fast Audio and Video encoder
usage: ffmpeg [options] [[infile options] -i infile]... {[outfile options] outfile}..
```

`ffmpeg -i narwhals.mp4 -c:v libx264 -c:a copy newnarwhals.mp4`

Then I uploaded it and got the same error:

```
D, [2016-07-13T16:11:45.340243 #1] DEBUG -- : ffmpeg -y  -i /tmp/moviemasher/download/e388d6b2bd2aa60638857fd9e29e68df7ec20402cf5c48b8350f3f8b0984e100/downloaded-intermediate.wav -i /tmp/moviemasher/download/e388d6b2bd2aa60638857fd9e29e68df7ec20402cf5c48b8350f3f8b0984e100/downloaded.mp4 -filter_complex "color=color=0x000000:duration=66.7:size=512x288:rate=30[layer0];[0:v]trim=duration=66.7:start=2.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1,drawtext=fontcolor=0xff0000@1.0:shadowcolor=0x000000@0.0:fontsize=41:x=216:y=-2:shadowx=8:shadowy=4:fontfile=/tmp/moviemasher/download/4963dad43ca77bb16a721c3690198ec900597b13222d32505e05fcd087992e68/downloaded.ttf:textfile=/tmp/moviemasher/render/34dadb90-168f-462c-9ab9-7f1a3176b5f5/61e3ded7-f4a7-48c8-afa6-63d01d0a83f1/db371418-3b8a-4965-98f0-d0e879c96368.txt[layer1];[layer0][layer1]overlay=x=0:y=0" -af \[0:a\]atrim\=start\=2.0:duration\=66.7,asetpts\=expr\=PTS-STARTPTS -async 1 -t 66.7 -b:a 224k -r:a 44100 -c:a aac -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\=new /tmp/moviemasher/render/34dadb90-168f-462c-9ab9-7f1a3176b5f5/61e3ded7-f4a7-48c8-afa6-63d01d0a83f1.mp4
D, [2016-07-13T16:11:45.378655 #1] DEBUG -- : ffmpeg version 3.0.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.9.2 (Debian 4.9.2-10)
  configuration: --enable-frei0r --enable-gpl --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libxvid --enable-postproc --enable-pthreads --enable-version3 --enable-zlib --extra-cflags=-I/usr/local/include/openjpeg
  libavutil      55. 17.103 / 55. 17.103
  libavcodec     57. 24.102 / 57. 24.102
  libavformat    57. 25.100 / 57. 25.100
  libavdevice    57.  0.101 / 57.  0.101
  libavfilter     6. 31.100 /  6. 31.100
  libswscale      4.  0.100 /  4.  0.100
  libswresample   2.  0.101 /  2.  0.101
  libpostproc    54.  0.100 / 54.  0.100
Guessed Channel Layout for  Input Stream #0.0 : stereo
Input #0, wav, from '/tmp/moviemasher/download/e388d6b2bd2aa60638857fd9e29e68df7ec20402cf5c48b8350f3f8b0984e100/downloaded-intermediate.wav':
  Metadata:
    encoder         : Lavf57.25.100
  Duration: 00:01:08.80, bitrate: 1411 kb/s
    Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 44100 Hz, 2 channels, s16, 1411 kb/s
Input #1, mov,mp4,m4a,3gp,3g2,mj2, from '/tmp/moviemasher/download/e388d6b2bd2aa60638857fd9e29e68df7ec20402cf5c48b8350f3f8b0984e100/downloaded.mp4':
  Metadata:
    major_brand     : isom
    minor_version   : 512
    compatible_brands: isomiso2avc1mp41
    encoder         : Lavf56.40.101
  Duration: 00:01:08.80, start: 0.000000, bitrate: 2725 kb/s
    Stream #1:0(und): Video: h264 (High) (avc1 / 0x31637661), yuv420p, 1280x720 [SAR 1:1 DAR 16:9], 2528 kb/s, 25 fps, 25 tbr, 12800 tbn, 50 tbc (default)
    Metadata:
      handler_name    : VideoHandler
    Stream #1:1(und): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 191 kb/s (default)
    Metadata:
      handler_name    : SoundHandler
Stream specifier ':v' in filtergraph description color=color=0x000000:duration=66.7:size=512x288:rate=30[layer0];[0:v]trim=duration=66.7:start=2.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1,drawtext=fontcolor=0xff0000@1.0:shadowcolor=0x000000@0.0:fontsize=41:x=216:y=-2:shadowx=8:shadowy=4:fontfile=/tmp/moviemasher/download/4963dad43ca77bb16a721c3690198ec900597b13222d32505e05fcd087992e68/downloaded.ttf:textfile=/tmp/moviemasher/render/34dadb90-168f-462c-9ab9-7f1a3176b5f5/61e3ded7-f4a7-48c8-afa6-63d01d0a83f1/db371418-3b8a-4965-98f0-d0e879c96368.txt[layer1];[layer0][layer1]overlay=x=0:y=0 matches no streams.
```

**Comment by alvaromlg** [comment]
I figured out how to reproduce this bug, I think it doesnt matter the encoding.

If you add a video and then a watermark (an image layer), if that watermark has the same duration as the video duration (because you want the watermark be present during all the playtime) it will crash with the error described above.

So if the length of the video is 1200 then the length of the watermark must be differnet from 1200. In my testing it works with 1201 or 1199.

That's why sometimes it was working and after playing with the duration it would stop working.

The strange thing is the pipeline is very different in the working case and in the other one.

Here it is crashing with the Stream specifier doesnt match error with an image and a video woth with duration 1200

`ffmpeg -y  -i /tmp/moviemasher/download/bc06ff974916645404c707737e871a792e488e1e86ad965c12dac900c0414327/downloaded-intermediate.wav -i /tmp/moviemasher/download/bc06ff974916645404c707737e871a792e488e1e86ad965c12dac900c0414327/downloaded.mp4 -loop 1 -i /tmp/moviemasher/download/ef2dc68b64620d7dc264becad392d41b5e9e0ea4edd561f3234a81ef303fb9bb/downloaded.png -filter_complex \"color=color=0x000000:duration=120.0:size=512x288:rate=30[layer0];[0:v]trim=duration=120.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1,colorchannelmixer=rr=0.3:rg=0.4:rb=0.3:ra=0:gr=0.3:gg=0.4:gb=0.3:ga=0:br=0.3:bg=0.4:bb=0.3:ba=0:ar=0:ag=0:ab=0:aa=1,drawtext=fontcolor=0xff0000@1.0:shadowcolor=0x000000@0.0:fontsize=28:x=191:y=0:shadowx=8:shadowy=4:fontfile=/tmp/moviemasher/download/4963dad43ca77bb16a721c3690198ec900597b13222d32505e05fcd087992e68/downloaded.ttf:textfile=/tmp/moviemasher/render/f84264ab-de80-4f36-951f-9ec275dd83e0/78ba6c68-22de-4f79-b785-2bf1db6acf78/6d81d6fc-d92b-49f5-97fa-a58714667ab2.txt,chromakey=color=0x00ff00:blend=0.01:similarity=0.5[layer1];[1:v]setpts=expr=PTS-STARTPTS,scale=width=105.072:height=57.312,setsar=sar=1:max=1[layer2];[layer0][layer1]overlay=x=0:y=0[layered1];[layered1][layer2]overlay=x=0:y=0\" -af \\[0:a\\]atrim\\=start\\=0.0:duration\\=120.0,asetpts\\=expr\\=PTS-STARTPTS -async 1 -t 120.0 -b:a 224k -r:a 44100 -c:a aac -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\\=transcoded /tmp/moviemasher/render/f84264ab-de80-4f36-951f-9ec275dd83e0/78ba6c68-22de-4f79-b785-2bf1db6acf78.mp4`

And this one is workiing, with the exact same mash configuration, video 1200 duration, image 1201 duration:

`ffmpeg -y -i /tmp/moviemasher/download/bc06ff974916645404c707737e871a792e488e1e86ad965c12dac900c0414327/downloaded.mp4 -loop 1 -i /tmp/moviemasher/download/ef2dc68b64620d7dc264becad392d41b5e9e0ea4edd561f3234a81ef303fb9bb/downloaded.png -filter_complex \"color=color=0x000000:duration=120.0:size=512x288:rate=30[layer0];[0:v]trim=duration=120.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1,colorchannelmixer=rr=0.3:rg=0.4:rb=0.3:ra=0:gr=0.3:gg=0.4:gb=0.3:ga=0:br=0.3:bg=0.4:bb=0.3:ba=0:ar=0:ag=0:ab=0:aa=1,drawtext=fontcolor=0xff0000@1.0:shadowcolor=0x000000@0.0:fontsize=28:x=191:y=0:shadowx=8:shadowy=4:fontfile=/tmp/moviemasher/download/4963dad43ca77bb16a721c3690198ec900597b13222d32505e05fcd087992e68/downloaded.ttf:textfile=/tmp/moviemasher/render/5d74f630-3a50-4f56-b4bf-96a34b8f3504/73ea47e0-a5fd-4ede-ba0b-66cd0cd69eb9/839fb338-ec0d-44b6-a86b-9b3a9e3f837f.txt,chromakey=color=0x00ff00:blend=0.01:similarity=0.5[layer1];[1:v]setpts=expr=PTS-STARTPTS,scale=width=105.072:height=57.312,setsar=sar=1:max=1,trim=duration=120.0,setpts=expr=PTS-STARTPTS[layer2];[layer0][layer1]overlay=x=0:y=0[layered1];[layered1][layer2]overlay=x=0:y=0\" -t 120.0 -an -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\\=transcoded -qp 0 /tmp/moviemasher/render/5d74f630-3a50-4f56-b4bf-96a34b8f3504/73ea47e0-a5fd-4ede-ba0b-66cd0cd69eb9/concat-0.mp4`

There are a lot of differences in the pipeline just increasing the watermark image duration from 1200 to 1201

The first one is the resulting video is always called concat something, for example concat-0.mp4 in this working example. When it fails it's always called with some generated hash like 78ba6c68-22de-4f79-b785-2bf1db6acf78.mp4

Also another noticable difference is in the working example it adds trim=duration=120.0 in the filter_complex parameter and not in the not working example.

Also the not working example adds this trim outside the filter_complex: \]atrim\\=start\\=0.0:duration\\=120.0,asetpts\\=expr\\=PTS-STARTPTS -async 1

And well, there are more different flags... so I am not sure why they both are so different or even if both pipelines have the same prupose.

I am comparing the not working ffmpeg pipeline logged in error/*/log.txt and angular-moviemasher.log with the success pipeline logged in angular-moviemasher.log when I press render.

**Comment by syntropo** [comment]
Yes, well, it seems you inadvertently ran the test I asked you to run even though I hadn't really phrased it properly - thanks for reading my mind :)

I still believe it is a problem with the encoding of the file. I'd be curious if using, for example, a new trailer from Apple would work in the original case. I believe the second case is working because all that concat magic is ultimately re-encoding your file at a low level. I'm not sure the copy channel command you used would do the trick.

**Comment by alvaromlg** [comment]
I tried with several different .mp4 files. I will try as soon as possible with an Apple trailer!

**Comment by alvaromlg** [comment]
Hello!

I tried with an Apple's trailer. It's an m4v file. This one:

```
ffprobe outlawsandangels-clp_i320.m4v
ffprobe version 2.8.6 Copyright (c) 2007-2016 the FFmpeg developers
  built with gcc 5.1.1 (GCC) 20150618 (Red Hat 5.1.1-4)
  configuration: --prefix=/usr --bindir=/usr/bin --datadir=/usr/share/ffmpeg --incdir=/usr/include/ffmpeg --libdir=/usr/lib64 --mandir=/usr/share/man --arch=x86_64 --optflags='-O2 -g -pipe -Wall -Werror=format-security -Wp,-D_FORTIFY_SOURCE=2 -fexceptions -fstack-protector-strong --param=ssp-buffer-size=4 -grecord-gcc-switches -specs=/usr/lib/rpm/redhat/redhat-hardened-cc1 -m64 -mtune=generic' --enable-bzlib --disable-crystalhd --enable-frei0r --enable-gnutls --enable-ladspa --enable-libass --enable-libcdio --enable-libdc1394 --disable-indev=jack --enable-libfreetype --enable-libgsm --enable-libmp3lame --enable-openal --enable-libopencv --enable-libopenjpeg --enable-libopus --enable-libpulse --enable-libschroedinger --enable-libsoxr --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libv4l2 --enable-libvpx --enable-libx264 --enable-libx265 --enable-libxvid --enable-x11grab --enable-avfilter --enable-avresample --enable-postproc --enable-pthreads --disable-static --enable-shared --enable-gpl --disable-debug --disable-stripping --shlibdir=/usr/lib64 --enable-runtime-cpudetect
  libavutil      54. 31.100 / 54. 31.100
  libavcodec     56. 60.100 / 56. 60.100
  libavformat    56. 40.101 / 56. 40.101
  libavdevice    56.  4.100 / 56.  4.100
  libavfilter     5. 40.101 /  5. 40.101
  libavresample   2.  1.  0 /  2.  1.  0
  libswscale      3.  1.101 /  3.  1.101
  libswresample   1.  2.101 /  1.  2.101
  libpostproc    53.  3.100 / 53.  3.100
Input #0, mov,mp4,m4a,3gp,3g2,mj2, from 'outlawsandangels-clp_i320.m4v':
  Metadata:
    major_brand     : M4V 
    minor_version   : 1
    compatible_brands: M4V M4A mp42isom
    creation_time   : 2016-07-01 15:23:50
  Duration: 00:01:46.02, start: 0.000000, bitrate: 1597 kb/s
    Stream #0:0(eng): Video: h264 (Constrained Baseline) (avc1 / 0x31637661), yuv420p(tv, smpte170m/smpte170m/bt709), 640x480 [SAR 2000:1500 DAR 16:9], 1488 kb/s, 23.98 fps, 23.98 tbr, 24k tbn, 48k tbc (default)
    Metadata:
      creation_time   : 2016-07-01 15:23:50
      handler_name    : Apple Video Media Handler
    Stream #0:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 106 kb/s (default)
    Metadata:
      creation_time   : 2016-07-01 15:23:50
      handler_name    : Apple Sound Media Handler
```

This time I tried to just trim it directly. Without using watermarks like in my previous attempts.

I added 20.0 secs in the Trim input and 700 secs in the Length input to trim it at the beginning and at the end, in both sides.

Then the transcoding crashed with this error:

It looks like a very similar error that with the -mp4 files:

```
couldn't create /tmp/moviemasher/render/3513d19e-4c88-4dd3-a54d-8ffbcc802d80/5c5474ff-3cc8-4a9a-8af8-675a26cbeab2.mp4
ffmpeg -y  -i /tmp/moviemasher/download/aec81c5487e38b39c544cdccdde36e27176f3616df7dbbad983ce4ebeb4e285f/downloaded-intermediate.wav -i /tmp/moviemasher/download/aec81c5487e38b39c544cdccdde36e27176f3616df7dbbad983ce4ebeb4e285f/downloaded.m4v -filter_complex "color=color=0x000000:duration=70.0:size=512x288:rate=30[layer0];[0:v]trim=duration=70.0:start=20.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0" -af \[0:a\]atrim\=start\=20.0:duration\=70.0,asetpts\=expr\=PTS-STARTPTS -async 1 -t 70.0 -b:a 224k -r:a 44100 -c:a aac -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\=apple\ trailer -pass 2 -passlogfile /tmp/moviemasher/render/3513d19e-4c88-4dd3-a54d-8ffbcc802d80/pass-c9bd7315-cc7f-4787-bc42-72ac09930e5d /tmp/moviemasher/render/3513d19e-4c88-4dd3-a54d-8ffbcc802d80/5c5474ff-3cc8-4a9a-8af8-675a26cbeab2.mp4
W, [2016-07-19T09:23:27.207423 #1]  WARN -- : two pass encoding failed, retrying in one
D, [2016-07-19T09:23:27.207505 #1] DEBUG -- : ffmpeg -y  -i /tmp/moviemasher/download/aec81c5487e38b39c544cdccdde36e27176f3616df7dbbad983ce4ebeb4e285f/downloaded-intermediate.wav -i /tmp/moviemasher/download/aec81c5487e38b39c544cdccdde36e27176f3616df7dbbad983ce4ebeb4e285f/downloaded.m4v -filter_complex "color=color=0x000000:duration=70.0:size=512x288:rate=30[layer0];[0:v]trim=duration=70.0:start=20.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0" -af \[0:a\]atrim\=start\=20.0:duration\=70.0,asetpts\=expr\=PTS-STARTPTS -async 1 -t 70.0 -b:a 224k -r:a 44100 -c:a aac -s 512x288 -c:v libx264 -level 41 -movflags faststart -b:v 2000k -r:v 30 -metadata title\=apple\ trailer /tmp/moviemasher/render/3513d19e-4c88-4dd3-a54d-8ffbcc802d80/5c5474ff-3cc8-4a9a-8af8-675a26cbeab2.mp4
D, [2016-07-19T09:23:27.244825 #1] DEBUG -- : ffmpeg version 3.0.2 Copyright (c) 2000-2016 the FFmpeg developers
  built with gcc 4.9.2 (Debian 4.9.2-10)
  configuration: --enable-frei0r --enable-gpl --enable-libfontconfig --enable-libfreetype --enable-libfribidi --enable-libmp3lame --enable-libopencore-amrnb --enable-libopencore-amrwb --enable-libopenjpeg --enable-libspeex --enable-libtheora --enable-libvorbis --enable-libvpx --enable-libx264 --enable-libxvid --enable-postproc --enable-pthreads --enable-version3 --enable-zlib --extra-cflags=-I/usr/local/include/openjpeg
  libavutil      55. 17.103 / 55. 17.103
  libavcodec     57. 24.102 / 57. 24.102
  libavformat    57. 25.100 / 57. 25.100
  libavdevice    57.  0.101 / 57.  0.101
  libavfilter     6. 31.100 /  6. 31.100
  libswscale      4.  0.100 /  4.  0.100
  libswresample   2.  0.101 /  2.  0.101
  libpostproc    54.  0.100 / 54.  0.100
Guessed Channel Layout for  Input Stream #0.0 : stereo
Input #0, wav, from '/tmp/moviemasher/download/aec81c5487e38b39c544cdccdde36e27176f3616df7dbbad983ce4ebeb4e285f/downloaded-intermediate.wav':
  Metadata:
    encoder         : Lavf57.25.100
  Duration: 00:01:46.05, bitrate: 1411 kb/s
    Stream #0:0: Audio: pcm_s16le ([1][0][0][0] / 0x0001), 44100 Hz, 2 channels, s16, 1411 kb/s
Input #1, mov,mp4,m4a,3gp,3g2,mj2, from '/tmp/moviemasher/download/aec81c5487e38b39c544cdccdde36e27176f3616df7dbbad983ce4ebeb4e285f/downloaded.m4v':
  Metadata:
    major_brand     : M4V
    minor_version   : 1
    compatible_brands: M4V M4A mp42isom
    creation_time   : 2016-07-01 15:23:50
  Duration: 00:01:46.02, start: 0.000000, bitrate: 1597 kb/s
    Stream #1:0(eng): Video: h264 (Constrained Baseline) (avc1 / 0x31637661), yuv420p(tv, smpte170m/smpte170m/bt709), 640x480 [SAR 2000:1500 DAR 16:9], 1488 kb/s, 23.98 fps, 23.98 tbr, 24k tbn, 48k tbc (default)
    Metadata:
      creation_time   : 2016-07-01 15:23:50
      handler_name    : Apple Video Media Handler
    Stream #1:1(eng): Audio: aac (LC) (mp4a / 0x6134706D), 44100 Hz, stereo, fltp, 106 kb/s (default)
    Metadata:
      creation_time   : 2016-07-01 15:23:50
      handler_name    : Apple Sound Media Handler
Stream specifier ':v' in filtergraph description color=color=0x000000:duration=70.0:size=512x288:rate=30[layer0];[0:v]trim=duration=70.0:start=20.0,fps=fps=30,setpts=expr=PTS-STARTPTS,scale=width=512:height=288,setsar=sar=1:max=1[layer1];[layer0][layer1]overlay=x=0:y=0 matches no streams.
```

Maybe I am using the mash editor wrongly, I just wanted to, for example, cut off 20 seconds at the beginning and 20 seconds at the ending.

Also I dont know if that's related but I installed it with docker. I ran:

`docker run -d -p 8080:80 --name=angular_moviemasher moviemasher/angular-moviemasher`

and

`docker run -d -t --volumes-from=angular_moviemasher --name=moviemasher_rb moviemasher/moviemasher.rb process_loop`

The host machine is a CentOS 7

**Comment by syntropo** [comment]
Okay, thanks to your great bug report it seems I was able to get to the root of the problem. Evidently, FFmpeg is expecting [1:v] because it's counting the nonexistent video channel in the audio file. To fix this I've simply reversed the order of the inputs so that the audio is always second. 

I've just pushed version 4.0.19 here on Github and the image in Docker Hub should complete building within the hour. Don't forget to delete your current local image.
